scvid/train/training_plan.py

```diff
@@ -28,43 +25,57 @@ class PyroTrainingPlan(pl.LightningModule):
 
     def __init__(
         self,
-        pyro_module: pyro.nn.PyroModule,
-        loss_fn: pyro.infer.ELBO | None = None,
-        optim_fn: Callable | None = None,
+        module: BaseModule,
+        optim_fn: type[torch.optim.Optimizer] | str | None = None,
         optim_kwargs: dict | None = None,
-        scheduler_fn: Callable | None = None,
+        scheduler_fn: type[torch.optim.lr_scheduler.LRScheduler] | str | None = None,
         scheduler_kwargs: dict | None = None,
-    ):
+    ) -> None:
         super().__init__()
-        self.module = pyro_module
+        self.module = module
+
+        # set up optimizer and scheduler
+        if isinstance(optim_fn, str):
+            class_module, class_name = optim_fn.rsplit(".", 1)
+            optim_module = import_module(class_module)
+            self.optim_fn = getattr(optim_module, class_name)
+        elif optim_fn is None:
+            self.optim_fn = torch.optim.Adam
+        else:
+            self.optim_fn = optim_fn
+
+        if isinstance(scheduler_fn, str):
+            class_module, class_name = scheduler_fn.rsplit(".", 1)
+            scheduler_module = import_module(class_module)
+            self.scheduler_fn = getattr(scheduler_module, class_name)
+        else:
+            self.scheduler_fn = scheduler_fn
 
-        self.optim_fn = torch.optim.Adam if optim_fn is None else optim_fn
         optim_kwargs = {} if optim_kwargs is None else optim_kwargs
         if "lr" not in optim_kwargs:
             optim_kwargs["lr"] = 1e-3
```

Comment:
Magic number in the middle of a method-- move it to the initializer kwargs with a default or a _consts.py.